internal/command/parse.go

```diff
@@ -0,0 +1,131 @@
+package command
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+type Usage struct {
+	Flags []string
+}
+
+var commandMap map[string]*Usage
+var invalidFlags []string
+
+func ParseOpts() int {
+
+	commandMap = map[string]*Usage{
+		"score-compose": {
+			Flags: []string{"h", "v", "help", "version"},
```

Comment:
It's going to be really painful if we have to keep this structure synchronised with the cobra cli definitions. Have you considered using https://pkg.go.dev/github.com/spf13/cobra#Command.SetFlagErrorFunc to get a hook on the particular command that failed validation?

Then we can inspect Flags to work out which flags were expected here?

Reply:
Thanks, I'll check this out.